lib/haskell/natural4/src/LS/Interpreter.hs

```diff
@@ -865,11 +887,12 @@ expandBSRM l4i depth x = do
   pure toreturn
 
 -- | Do expansion, throwing away the LHS IS part of any `RPBoolStructR` elements we encounter.
+-- However, do keep the title of the old LHS in a label.
 expandBSR' :: Interpreted -> Int -> BoolStructR -> BoolStructR
 expandBSR' l4i depth = \case
   AA.Leaf rp -> case expandRP l4i depth1 rp of
-    RPBoolStructR _mt1 RPis bsr -> bsr
-    o                           -> AA.mkLeaf o
+    RPBoolStructR lhs RPis bsr -> unleaf' (Just $ AA.Metadata $ mt2text lhs) bsr
+    o                          -> AA.mkLeaf o
```

Comment:
Is this a bug fix? Like, was the simplification performed by `unleaf` not applied before that? Also, it looks like `expandRP` traverses the full `rp`, calling `expandBSR` again. There seems to be a considerable overlap with the implementation of `unleaf'` as well.